arrebol/worker/manager/manager.go

```diff
@@ -0,0 +1,34 @@
+package manager
+
+import (
+	"github.com/ufcg-lsd/arrebol-pb/arrebol/worker"
+	"github.com/ufcg-lsd/arrebol-pb/storage"
+)
+
+type Manager struct {
+	storage *storage.Storage
+}
+
+func NewManager(storage *storage.Storage) *Manager {
+	return &Manager{
+		storage:storage,
+	}}
+
+func (m *Manager) Join(w worker.Worker) (uint, error) {
+	queueId := m.selectQueue(w)
+	w.QueueID = queueId
+	queue, err := m.storage.RetrieveQueue(queueId)
+	if err != nil {
+		return 0, err
+	}
+	queue.Workers = append(queue.Workers, &w)
+	err = m.storage.SaveQueue(queue)
+	if err != nil {
```

Comment:
`0` means we have problems? I think we are missing docs to this fuction

Reply:
What it means that we have a problem is if the err variable is not nil. 0 is just the zero value for the uint type. I added a doc to this function.